wasm.js

```diff
@@ -0,0 +1,174 @@
+        console.error(err);
+        throw err
+    }
+
+    return sig.serializeToHexStr();
+}
+
+// Initialize __zcn_wasm__ 
+if (!g.__zcn_wasm__) {
+    g.__zcn_wasm__ = {
+        bls: {
+            secretKey: null,
+            sign: blsSign,
+        },
+        sdk: {},
+    };
+}
+
+/**
+ * bridge is an easier way to refer to the Go WASM object.
+ */
+const bridge = g.__zcn_wasm__;
+
+/**
+ * Sleep is used when awaiting for Go Wasm to initialize.
+ * It uses the lowest possible sane delay time (via requestAnimationFrame).
+ * However, if the window is not focused, requestAnimationFrame never returns.
+ * A timeout will ensure to be called after 50 ms, regardless of whether or not the tab is in focus.
+ *
+ * @returns {Promise} an always-resolving promise when a tick has been completed.
+ */
+const sleep = () => new Promise(res => {
+    requestAnimationFrame(res);
+    setTimeout(res, 50);
+});
+
+/**
+ * The maximum amount of time that we would expect Wasm to take to initialize.
+ * If it doesn't initialize after this time, we send a warning to console.
+ * Most likely something has gone wrong if it takes more than 3 seconds to initialize.
+ */
+const maxTime = 3 * 1000;
+
+async function setWallet(bls, clientID, sk, pk) {
+    bridge.bls.secretKey = bls.deserializeHexStrToSecretKey(sk);
+    await bridge.sdk.setWallet(clientID, pk)
+}
+
+const blsProxy = new Proxy(
+    {},
+    {
+        get: (_, key) => bridge.bls[key],
+        set: (_, key, value) => {
+            bridge.bls[key] = value;
+        },
+    }
+)
+
+async function createWasm() {
+    const go = new g.Go();
+
+    const result = await WebAssembly.instantiateStreaming(
+        await fetch('/zcn.wasm'),
+        go.importObject
+    );
+
+    go.run(result.instance);
+
+    setTimeout(() => {
+        if (bridge.__wasm_initialized__ !== true) {
+            console.warn(
+                'wasm bridge (zcn.__wasm_initialized__) still not true after max time'
+            );
+        }
+    }, maxTime);
+
+    const sdkGet = (_, key) => (...args) => new Promise(async (res, rej) => {
+        if (!go || go.exited) {
+            return rej(new Error('The Go instance is not active.'));
+        }
+
+        while (bridge.__wasm_initialized__ !== true) {
+            await sleep();
+        }
+
+        if (typeof bridge.sdk[key] !== 'function') {
+            res(bridge.sdk[key]);
+
+            if (args.length !== 0) {
+                console.warn(
+                    'Retrieved value from WASM returned function type, however called with arguments.'
+                );
+            }
+            return;
+        }
+
+        try {
+            let resp = bridge.sdk[key].apply(undefined, args);
+
+            // support wasm.BindAsyncFunc
+
+            if (resp && typeof resp.then === 'function') {
+                resp = await Promise.race([resp]);
+            }
+
+            if (resp?.error) {
+                rej(resp.error);
```

Comment:
It will be nice to follow the same pattern used on line 114:

```suggestion
                rej(new Error(resp.error));
```